### PR TITLE
common: Make sure + - * / % binary operators work the same in all dialects

### PR DESCRIPTION
**Pull request by lustefaniak**
After https://github.com/sqlparser-rs/sqlparser-rs/pull/1009 is merged this will pass.

From what I see in BigQuery docs, `da-sh-es` is invalid identifier when used without quotes.
https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#unquoted_identifiers

### CODE REVIEW

File: tests/sqlparser_bigquery.rs [hunk 1]
```diff
@@ -615,8 +615,6 @@ fn parse_table_identifiers() {
         assert!(bigquery().parse_sql_statements(&sql).is_err());
     }
 
-    test_table_ident("da-sh-es", None, vec![Ident::new("da-sh-es")]);
```
Comment (alamb) [inline tests/sqlparser_bigquery.rs:618]: Why was this removed?

**Review by alamb — COMMENTED** [review]
Thank you very much @lustefaniak  -- I have one question about the changed test, but otherwise looks good to me

I ran the tests locally after merging up from main and they worked great!

File: tests/sqlparser_bigquery.rs (hunk 1 above)
Comment (lustefaniak) [reply tests/sqlparser_bigquery.rs:618]: It's in the description: 

From what I see in BigQuery docs, da-sh-es is invalid identifier when used without quotes.
https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#unquoted_identifiers

Comment (lustefaniak) [reply tests/sqlparser_bigquery.rs:618]:
```
Unquoted identifiers
 - Must begin with a letter or an underscore (_) character.
 - Subsequent characters can be letters, numbers, or underscores (_).
```
So the existing code was wrong, as `da-sh-es` is `da - sh - es`

**Review by alamb — APPROVED** [review]
Thank you @lustefaniak
